inst/include/cpp11/protect.hpp

```diff
@@ -195,23 +198,27 @@ constexpr struct protect safe = {};
 inline void check_user_interrupt() { safe[R_CheckUserInterrupt](); }
 
 template <typename... Args>
-void stop [[noreturn]] (const char* fmt, Args... args) {
-  safe.noreturn(Rf_errorcall)(R_NilValue, fmt, args...);
+void stop [[noreturn]] (const char* fmt_arg, Args... args) {
+  std::string msg = fmt::format(fmt_arg, args...);
+  safe.noreturn(Rf_errorcall)(R_NilValue, "%s", msg.c_str());
 }
 
 template <typename... Args>
-void stop [[noreturn]] (const std::string& fmt, Args... args) {
-  safe.noreturn(Rf_errorcall)(R_NilValue, fmt.c_str(), args...);
+void stop [[noreturn]] (const std::string& fmt_arg, Args... args) {
+  std::string msg = fmt::format(fmt_arg, args...);
```

Comment:
Also, it's unfortunate that we're losing the `%x` format specifier for `SEXP`s. This could be recoverable by introducing a helper function:
```c++
// pass through by default
template <typename T>
const T& as_formattable(const T& value) { return value; }

// specialization for SEXP: 
std::string as_formattable(SEXP sexp) { return gettextf("%x", sexp); }
```

Reply:
I think you would use `cpp11::stop("{0:x}", my_sexp)` if you want to print the hexadecimal address for a sexp.

Reply:
Oh, I misunderstood the purpose of `%x`. I thought it gave a string representation of the pointed value rather than a representation of the pointer itself. Let me rephrase: would it be desirable to intercept `SEXP` pointers and replace them with `capture.output(print(value))` or so? It'd probably be better to include that as a `sexp::to_string()` so that we can write

```c++
cpp11::stop("Invalid: {}", sexp(value).to_string());
```

Reply:
Ah, maybe, though it is not always clear what output would be preferred for a given SEXP, possibly something like that returned from `R_inspect()` might be the most useful for any SEXP type, though the default arguments can generate a lot of data for deeply nested objects.

```r
> .Internal(inspect(1:10))
@7ff6163c1388 13 INTSXP g0c0 [REF(65535)]  1 : 10 (compact)
```